x-pack/plugins/task_manager/index.ts

```diff
@@ -26,19 +31,33 @@ export function taskManager(kibana) {
           .description('The name of the index used to store task information.')
           .default('.kibana_task_manager'),
         max_workers: Joi.number()
-          .description('The maximum number of tasks that this Kibana instance will run simultaneously.')
+          .description(
+            'The maximum number of tasks that this Kibana instance will run simultaneously.'
+          )
           .min(1) // disable the task manager rather than trying to specify it with 0 workers
           .default(10),
         override_num_workers: Joi.object()
           .pattern(/.*/, Joi.number().greater(0))
-          .description('Customize the number of workers occupied by specific tasks (e.g. override_num_workers.reporting: 2)')
-          .default({})
+          .description(
+            'Customize the number of workers occupied by specific tasks (e.g. override_num_workers.reporting: 2)'
+          )
+          .default({}),
       }).default();
     },
-    init(server) {
+    init(server: Legacy.Server) {
+      const { plugins } = createShim(server);
       const config = server.config();
-      const taskManager = new TaskManager(this.kbnServer, server, config);
-      server.decorate('server', 'taskManager', taskManager);
+      const services = {
+        log: (msg: any) => server.log(msg),
+        elasticsearch: plugins.elasticsearch,
+      };
+      const taskManagerInstance = new TaskManager(
+        services,
+        config,
+        this.kbnServer.afterPluginsInit
+      );
+      // @ts-ignore
+      server.decorate('server', 'taskManager', taskManagerInstance);
```

Comment:
I guess this is here because things that use the task manager access it via the legacy server object?  So we'll have to refactor this when everything moves over.

Reply:
I ended up doing the refactor in this PR :)